packages/ui/src/providers/RouteTransition/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useCallback, useEffect, useOptimistic, useRef } from 'react'
```

Comment:
Add 'use client' so that react compiler can run here. Same for LoadingBar/index.tsx